modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v560/migrator/ClaimDataMigrator.java

```diff
@@ -0,0 +1,296 @@
+import org.wso2.carbon.identity.claim.metadata.mgt.dao.CacheBackedLocalClaimDAO;
+import org.wso2.carbon.identity.claim.metadata.mgt.dao.ClaimDialectDAO;
+import org.wso2.carbon.identity.claim.metadata.mgt.dao.ExternalClaimDAO;
+import org.wso2.carbon.identity.claim.metadata.mgt.dao.LocalClaimDAO;
+import org.wso2.carbon.identity.claim.metadata.mgt.exception.ClaimMetadataException;
+import org.wso2.carbon.identity.claim.metadata.mgt.model.AttributeMapping;
+import org.wso2.carbon.identity.claim.metadata.mgt.model.ClaimDialect;
+import org.wso2.carbon.identity.claim.metadata.mgt.model.ExternalClaim;
+import org.wso2.carbon.identity.claim.metadata.mgt.model.LocalClaim;
+import org.wso2.carbon.identity.claim.metadata.mgt.util.ClaimConstants;
+import org.wso2.carbon.identity.core.migrate.MigrationClientException;
+import org.wso2.carbon.is.migration.internal.ISMigrationServiceDataHolder;
+import org.wso2.carbon.is.migration.service.Migrator;
+import org.wso2.carbon.is.migration.service.v540.util.FileBasedClaimBuilder;
+import org.wso2.carbon.is.migration.util.Constant;
+import org.wso2.carbon.is.migration.util.Utility;
+import org.wso2.carbon.user.api.Tenant;
+import org.wso2.carbon.user.api.UserRealm;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.user.core.UserCoreConstants;
+import org.wso2.carbon.user.core.claim.ClaimMapping;
+import org.wso2.carbon.user.core.claim.inmemory.ClaimConfig;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.xml.stream.XMLStreamException;
+
+/**
+ * This class handles the claim data migration.
+ */
+public class ClaimDataMigrator extends Migrator {
+
+    private static final Log log = LogFactory.getLog(ClaimDataMigrator.class);
+
+    private static final String CLAIM_CONFIG = "claim-config.xml";
+
+    private ClaimConfig claimConfig;
+
+    private ClaimDialectDAO claimDialectDAO = new ClaimDialectDAO();
+
+    private CacheBackedLocalClaimDAO localClaimDAO = new CacheBackedLocalClaimDAO(new LocalClaimDAO());
+
+    private CacheBackedExternalClaimDAO externalClaimDAO = new CacheBackedExternalClaimDAO(new ExternalClaimDAO());
+
+    @Override
+    public void migrate() throws MigrationClientException {
+
+        String filePath = Utility.getDataFilePath(CLAIM_CONFIG, getVersionConfig().getVersion());
+        try {
+            claimConfig = FileBasedClaimBuilder.buildClaimMappingsFromConfigFile(filePath);
+        } catch (IOException | XMLStreamException | UserStoreException e) {
+            String message = "Error while building claims from config file";
+            if (isContinueOnError()) {
+                log.error(message, e);
+            } else {
+                throw new MigrationClientException(message, e);
+            }
+        }
+
+        if (claimConfig.getClaims().isEmpty()) {
```

Comment:
If this line is executed after Line#84, claimConfig  would not be initialized properly. Better to move this part in to try block.